src/maths/matrices/Matrix4.cpp

```diff
@@ -0,0 +1,351 @@
+#include <array>
+#include <cmath>
+#include <Matrix4.hpp>
+#include <sstream>
+#include <vector>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Constructors
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Construct a new Matrix4 object using an array of 16 float values.
+ *
+ * @param list An array of 16 float values to initialize the matrix with
+ */
+Matrix4::Matrix4(const std::array<float, 16>& list)
+{
+    for (int i = 0; i < 16; ++i)
+    {
+        _data[i] = list[i];
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Getters
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @return The data of the matrix
+ */
+[[nodiscard]] const float* Matrix4::getData() const
+{
+    return _data;
+}
+
+/**
+ * @return The first row of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getRowOne() const
+{
+    return Vector4(_data[0], _data[1], _data[2], _data[3]);
+}
+
+/**
+ * @return The second row of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getRowTwo() const
+{
+    return Vector4(_data[4], _data[5], _data[6], _data[7]);
+}
+
+/**
+ * @return The third row of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getRowThree() const
+{
+    return Vector4(_data[8], _data[9], _data[10], _data[11]);
+}
+
+/**
+ * @return The fourth row of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getRowFour() const
+{
+    return Vector4(_data[12], _data[13], _data[14], _data[15]);
+}
+
+/**
+ * @return The first column of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getColumnOne() const
+{
+    return Vector4(_data[0], _data[4], _data[8], _data[12]);
+}
+
+/**
+ * @return The second column of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getColumnTwo() const
+{
+    return Vector4(_data[1], _data[5], _data[9], _data[13]);
+}
+
+/**
+ * @return The third column of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getColumnThree() const
+{
+    return Vector4(_data[2], _data[6], _data[10], _data[14]);
+}
+
+/**
+ * @return The fourth column of the matrix
+ */
+[[nodiscard]] Vector4 Matrix4::getColumnFour() const
+{
+    return Vector4(_data[3], _data[7], _data[11], _data[15]);
+}
```

Comment:
T'aurais pas mieux fait de faire ca ?

```suggestion
/**
 * @param rowIndex The index of the row that we want to get
 * 
 * @return The row corresponding to the index
 */
[[nodiscard]] Vector4 Matrix4::getRow(int rowIndex) const
{
    return Vector4(_data[rowIndex * 4 + 0], _data[rowIndex * 4 + 1], _data[rowIndex * 4 + 2], _data[rowIndex * 4 + 3]);
}

/**
 * @param columnIndex The index of the column that we want to get
 * 
 * @return The column corresponding to the index
 */
[[nodiscard]] Vector4 Matrix4::getColumn(int columnIndex) const
{
    return Vector4(_data[0 + columnIndex], _data[4 + columnIndex], _data[8 + columnIndex], _data[12 + columnIndex]);
}
```